### PR TITLE
Fix inference for xgboost fitted with early-stopping

### PR DESCRIPTION
**Pull request by avolkov-intel**
# Description
Earlier there was a difference in results between xgboost predictions and predictions of our modelbuilder if xgboost was fitted using early-stopping parameter. This pull request is to fix this issue

### CODE REVIEW

**Comment by avolkov-intel** [comment]
@Alexsandruss can we merge it?

File: src/gbt_convertors.pyx [hunk 1]
```diff
@@ -152,7 +152,8 @@ def get_gbt_model_from_xgboost(booster: Any) -> Any:
     else:
         is_regression = True
 
-    n_iterations = int(len(trees_arr) / (n_classes if n_classes > 2 else 1))
+    n_iterations = booster.best_ntree_limit
```
Comment (razdoburdin) [inline src/gbt_convertors.pyx:155]: I think we should use [booster.best_iteration](https://xgboost.readthedocs.io/en/latest/python/python_api.html#xgboost.XGBRegressor.best_iteration), not 
booster.best_ntree_limit.
From [this part of the code](https://github.com/dmlc/xgboost/blob/663136aa08c00598d8b49adf5901e4cb2ce187da/python-package/xgboost/core.py#L1143) I concluded, that best_ntree_limit will be removed soon.

Please take into account that best_iteration is starts from 0, while best_ntree_limit starts from 1. Thus we will have a bias equals to 1.

**Comment by napetrov** [comment]
Also a broader question - can we add tests for this case?  we might do this as a second iteration after enabling fix due to release timelines

**Comment by avolkov-intel** [comment]
> Also a broader question - can we add tests for this case? we might do this as a second iteration after enabling fix due to release timelines

Yes, I think we should do this in different pull request to merge that one before release

**Comment by samir-nasibli** [comment]
> > Also a broader question - can we add tests for this case? we might do this as a second iteration after enabling fix due to release timelines
> 
> Yes, I think we should do this in different pull request to merge that one before release

I am expecting that we are testing the fix in the PR, so it is better to add simple test case in this PR as well.

**Comment by napetrov** [comment]
Manual testing is fine for this PR integration, but should be added anyway

**Review by razdoburdin — APPROVED** [review]
Looks good

**Pull request merged**